clang/test/CodeGenSYCL/Inputs/sycl.hpp

```diff
@@ -441,6 +441,15 @@ kernel_parallel_for(const KernelType &KernelFunc) {
   KernelFunc(id<Dims>());
 }
 
+template <typename KernelName, typename KernelType>
+ATTR_SYCL_KERNEL void kernel_parallel_for(const KernelType &kernelFunc) {
+#ifdef __SYCL_DEVICE_ONLY__
+  kernelFunc();
+#else
+  (void)kernelFunc;
```

Comment:
Is the else part necessary?

Reply:
I'm not sure the whole new routine makes sense. Is there `parallel_for` variant which doesn't need a range as a parameter (and therefore its kernel doesn't need `id` parameter or something)?

Reply:
UPD: Can't find one in the spec.

Reply:
@Fznamznon @premanandrao I included this function based on what we have here - [clang/test/SemaSYCL/Inputs/sycl.hpp](https://github.com/intel/llvm/blob/sycl/clang/test/SemaSYCL/Inputs/sycl.hpp#L300) , so I can have a test case for an operator()() call that doesn't take any arguments.

But after having looked at the actual header files, looks like there is no such version for `parallel_for`. We do have one for [kernel_single_task](https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/handler.hpp#L1164)

Any reason we have the `operator()()` under `SemaSYCL/Inputs/sycl.hpp` or is it just a mistake?

Reply:
> Any reason we have the operator()() under SemaSYCL/Inputs/sycl.hpp or is it just a mistake?

I think it is just a mistake. We have `single_task` to tesk `operator()()`.